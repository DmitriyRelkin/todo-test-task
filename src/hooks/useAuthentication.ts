import * as LocalAuthentication from 'expo-local-authentication';
import { useDispatch } from "react-redux";
import { setAuthenticationStatus } from "../store/slices/authenticationSlice";

const useAuthentication = () => {
  const dispatch = useDispatch();

  const onAuthenticate = async () => {
    try {
      const { success } = await LocalAuthentication.authenticateAsync();
      dispatch(setAuthenticationStatus(success));
    } catch (error) {
      console.error('Authentication error:', error);
    }
  };

  return {
    onAuthenticate,
  }
};

export default useAuthentication;
