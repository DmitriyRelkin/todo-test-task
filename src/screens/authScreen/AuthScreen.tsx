import React from 'react';
import { View, Text, TouchableOpacity} from 'react-native';
import { getAuthScreenStyle} from "./styles";
import useAuthentication from "../../hooks/useAuthentication";

const AuthScreen: React.FC = () => {
  const { container, btnStyle, btnText } = getAuthScreenStyle();
  const { onAuthenticate } = useAuthentication();

  return (
    <View style={container}>
      <TouchableOpacity testID="auth-button" style={btnStyle} onPress={onAuthenticate}>
        <Text style={btnText}>Authenticate</Text>
      </TouchableOpacity>
    </View>
  );
};

export default AuthScreen;
