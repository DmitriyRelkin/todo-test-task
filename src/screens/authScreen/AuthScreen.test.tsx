import React from 'react';
import renderer, { act } from 'react-test-renderer';
import AuthScreen from './AuthScreen';
import useAuthentication from '../../hooks/useAuthentication';

jest.mock('../../hooks/useAuthentication', () => ({
  __esModule: true,
  default: jest.fn(),
}));

describe('AuthScreen', () => {
  let onAuthenticate: any;

  beforeEach(() => {
    onAuthenticate = jest.fn();
    // @ts-ignore
    useAuthentication.mockImplementation(() => ({ onAuthenticate }));
  })
  it('renders the AuthScreen screen correctly', () => {
    const tree = renderer.create(
      <AuthScreen />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
  it('calls onAuthenticate when the button is pressed', () => {
    let component: any;
    act(() => {
      component = renderer.create(<AuthScreen />);
    });

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    const button = component.root.findByProps({ testID: 'auth-button' });

    act(() => {
      button.props.onPress();
    });

    expect(onAuthenticate).toHaveBeenCalled(); // Check if onAuthenticate was called
  });
});
