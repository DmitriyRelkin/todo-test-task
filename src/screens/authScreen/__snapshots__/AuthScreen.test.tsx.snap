// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthScreen calls onAuthenticate when the button is pressed 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#86a5aa",
        "borderRadius": 12,
        "justifyContent": "center",
        "opacity": 1,
        "padding": 10,
      }
    }
    testID="auth-button"
  >
    <Text
      style={
        {
          "color": "white",
          "fontSize": 14,
          "fontWeight": "600",
        }
      }
    >
      Authenticate
    </Text>
  </View>
</View>
`;

exports[`AuthScreen renders the AuthScreen screen correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "width": "100%",
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#86a5aa",
        "borderRadius": 12,
        "justifyContent": "center",
        "opacity": 1,
        "padding": 10,
      }
    }
    testID="auth-button"
  >
    <Text
      style={
        {
          "color": "white",
          "fontSize": 14,
          "fontWeight": "600",
        }
      }
    >
      Authenticate
    </Text>
  </View>
</View>
`;
